google/cloud/pubsub/samples/subscription_admin_samples.cc

```diff
@@ -0,0 +1,163 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "google/cloud/pubsub/admin/subscription_admin_client.h"
+#include "google/cloud/pubsub/admin/topic_admin_client.h"
+#include "google/cloud/pubsub/samples/pubsub_samples_common.h"
+#include "google/cloud/pubsub/subscription.h"
+#include "google/cloud/internal/getenv.h"
+#include "google/cloud/internal/random.h"
+#include "google/cloud/testing_util/example_driver.h"
+#include <sstream>
+
+namespace {
+
+using google::cloud::pubsub::examples::Cleanup;
+using SubscriptionAdminCommand =
+    std::function<void(google::cloud::pubsub_admin::SubscriptionAdminClient,
+                       std::vector<std::string> const&)>;
+
+google::cloud::testing_util::Commands::value_type
+CreateSubscriptionAdminCommand(std::string const& name,
+                               std::vector<std::string> const& arg_names,
+                               SubscriptionAdminCommand const& command) {
+  auto adapter = [=](std::vector<std::string> const& argv) {
+    if ((argv.size() == 1 && argv[0] == "--help") ||
+        argv.size() != arg_names.size()) {
+      std::ostringstream os;
+      os << name;
+      for (auto const& a : arg_names) {
+        os << " <" << a << ">";
+      }
+      throw google::cloud::testing_util::Usage{std::move(os).str()};
+    }
+    google::cloud::pubsub_admin::SubscriptionAdminClient client(
+        google::cloud::pubsub_admin::MakeSubscriptionAdminConnection());
+    command(std::move(client), std::move(argv));
+  };
+  return google::cloud::testing_util::Commands::value_type{name,
+                                                           std::move(adapter)};
+}
+
+void CreateSubscription(
+    google::cloud::pubsub_admin::SubscriptionAdminClient client,
+    std::vector<std::string> const& argv) {
+  //! [create-subscription]
```

Comment:
FYI: nothing is referencing these regions.

Reply:
Ack done